examples/gail_interaction_dataset.py

```diff
@@ -0,0 +1,111 @@
+from bark_ml.library_wrappers.lib_tf2rl.agents.gail_agent import BehaviorGAILAgent
+from bark_ml.library_wrappers.lib_tf2rl.runners.gail_runner import GAILRunner
+from bark_ml.library_wrappers.lib_tf2rl.load_expert_trajectories import load_expert_trajectories
+from bark_ml.behaviors.cont_behavior import BehaviorContinuousML
+from bark_ml.observers.nearest_state_observer import NearestAgentsObserver
+from bark_ml.library_wrappers.lib_tf2rl.scenario_generator_wrapper import *
+from bark_ml.evaluators.goal_reached import GoalReached
+
+FLAGS = flags.FLAGS
+flags.DEFINE_enum("mode",
+                  "visualize",
+                  ["train", "visualize", "evaluate"],
+                  "Mode the configuration should be executed in.")
+
+
+def run_configuration(argv):
+  params = ParameterServer(filename="examples/example_params/gail_params.json")
+
+  # Uncomment these to use the pretrained agents from https://github.com/GAIL-4-BARK/large_data_store
+  # The agents are automatically integrated using bazel together with the expert trajectories
+  # params["ML"]["GAILRunner"]["tf2rl"]["logdir"] = "../com_github_gail_4_bark_large_data_store/pretrained_agents/gail/merging"
+  # params["ML"]["GAILRunner"]["tf2rl"]["model_dir"] = "../com_github_gail_4_bark_large_data_store/pretrained_agents/gail/merging"
+
+  # When training a gail agent we add a suffix to the specified model and log dir to distinguish between training runs.
+  # If you want to visualize or evaluate using your locally trained gail agent, you have to specify the run to use.
+  # Therefore look into the directory specified in params["ML"]["GAILRunner"]["tf2rl"]["logdir"] and
+  # pick one of your runs with the naming scheme '<timestamp>_DDPG_GAIL'
+  # Add the name of the folder with the run to your:
+  # params["ML"]["GAILRunner"]["tf2rl"]["logdir"] and params["ML"]["GAILRunner"]["tf2rl"]["model_dir"]
+  # So if your model_dir and logdir were 'examples/gail_training' it becomes 'examples/gail_training/20200807T121018.454776_DDPG_GAIL' in the gail_params.json
+  # Alternatively append it as in the following lines:
+  # params["ML"]["GAILRunner"]["tf2rl"]["logdir"] = os.path.join(params["ML"]["GAILRunner"]["tf2rl"]["logdir"], "20200807T121018.454776_DDPG_GAIL")
+  # params["ML"]["GAILRunner"]["tf2rl"]["model_dir"] = os.path.join(params["ML"]["GAILRunner"]["tf2rl"]["model_dir"], "20200807T121018.454776_DDPG_GAIL")
+
+  # set you json config that contains a map and matching tracks.
+  scenario_generation = InteractionDatasetScenarioGeneration(num_scenarios=1,
+                                                             random_seed=0,
+                                                             params=params)
+  scenario = scenario_generation.create_scenarios(params, 1)[0]
+
+  sim_step_time = params["Scenario"]["Generation"][
+      "StepTime", "Step-time used in simulation", 0.2]
+  ego_agent_id = params["Scenario"]["Generation"][
```

Comment:
Is this the check if our agent 66 reach its goal and if yes, we should implement another criteria!

Reply:
Can you explain more what you mean? The ego_agent_id is just the id of the agent to replace in the simulation.